cmake/Modules/legate_core_options.cmake

```diff
@@ -0,0 +1,126 @@
+#=============================================================================
+# Copyright 2022 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+
+option(BUILD_SHARED_LIBS "Build legate.core shared libraries" ON)
+
+function(set_or_default var_name var_env)
+  list(LENGTH ARGN num_extra_args)
+  if(num_extra_args GREATER 0)
+    list(GET ARGN 0 var_default)
+  endif()
+  if(DEFINED ${var_name})
+    set(${var_name} ${${var_name}} PARENT_SCOPE)
```

Comment:
My brain might be failing here. What is the purpose of this line? Isn't it just defining the variable to the value it already has in the parent scope?

Reply:
Yep, that's right. It's probably unnecessary.